### DISCUSSION TITLE
visual-automata library does not work, same goes for colormath. it says - Couldn't find a pure Python 3 wheel for 'colormath'

### DISCUSSION THREAD

**Discussion by TG721** [pyscript/pyscript#1179] [Packages]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

If following libraries were supported, the code would run normally I presume, and my automata diagram would be displayed but it seems visual-automata library does not work/is not supported.

### What browsers are you seeing the problem on? (if applicable)

Chrome, Microsoft Edge

### Console info

```shell
Uncaught (in promise) PythonError: Traceback (most recent call last):
  File "/lib/python3.10/asyncio/futures.py", line 201, in result
    raise self._exception
  File "/lib/python3.10/asyncio/tasks.py", line 234, in __step
    result = coro.throw(exc)
  File "/lib/python3.10/site-packages/micropip/_micropip.py", line 183, in install
    transaction = await self.gather_requirements(requirements, ctx, keep_going)
  File "/lib/python3.10/site-packages/micropip/_micropip.py", line 173, in gather_requirements
    await gather(*requirement_promises)
  File "/lib/python3.10/asyncio/futures.py", line 284, in __await__
    yield self  # This tells Task to wait for completion.
  File "/lib/python3.10/asyncio/tasks.py", line 304, in __wakeup
    future.result()
  File "/lib/python3.10/asyncio/futures.py", line 201, in result
    raise self._exception
  File "/lib/python3.10/asyncio/tasks.py", line 232, in __step
    result = coro.send(None)
  File "/lib/python3.10/site-packages/micropip/_micropip.py", line 286, in add_requirement
    raise ValueError(
ValueError: Couldn't find a pure Python 3 wheel for 'colormath'. You can use `micropip.install(..., keep_going=True)` to get a list of all packages with missing wheels.

    at new_error (pyodide.asm.js:14:238191)
    at pyodide.asm.wasm:0xedbcb
    at pyodide.asm.wasm:0xf1a0e
    at method_call_trampoline (pyodide.asm.js:14:238105)
    at pyodide.asm.wasm:0x134c2c
    at pyodide.asm.wasm:0x217a84
    at pyodide.asm.wasm:0x174a14
    at pyodide.asm.wasm:0x135149
    at pyodide.asm.wasm:0x135243
    at pyodide.asm.wasm:0x1352e6
    at pyodide.asm.wasm:0x1fff83
    at pyodide.asm.wasm:0x1f98b5
    at pyodide.asm.wasm:0x135329
    at pyodide.asm.wasm:0x201f1b
    at pyodide.asm.wasm:0x1ff9ff
    at pyodide.asm.wasm:0x1f98b5
    at pyodide.asm.wasm:0x135329
    at pyodide.asm.wasm:0xf16d8
    at Module.callPyObjectKwargs (pyproxy.gen.ts:360:23)
    at Module.callPyObject (pyproxy.gen.ts:384:17)
    at wrapper (pyodide.asm.js:14:205222)
```

**Comment by nmstoker** [comment]
I haven't checked this, but i suspect you should take the error reason at face value (ie it really is going wrong because there's no pure Python wheel found) and if you resolve that you'll be fine.

If you take a look here, there are details of how to create a wheel yourself and then you should try importing that:
https://pyodide.org/en/stable/development/new-packages.html#determining-if-creating-a-pyodide-package-is-necessary

Btw we can guess at what your code did based on the traceback, but generally it's best all round if you can include it (or a minimally reproducible version if you can't post the full version)

**Comment by marimeireles** [comment]
This is happening because there isn't such a package in the pyodide ecosystem (which is what pyscript is using under the hood).

**Reply by TG721** [reply]
I checked pyodide packages. https://pyodide.org/en/stable/usage/packages-in-pyodide.html 
Interestingly [automata-lib](https://pypi.org/project/automata-lib/) library as I tested when I made my initial issue report worked fine with PyScript however [visual-automata](https://pypi.org/project/visual-automata/) library did not. Both packages are not part of pyodide packages list. so why does one package work with pyscript and the other does not? perhaps because the other makes GUI output and first just terminal one? I don't know
